cmd/osbuild-service-maintenance/aws.go

```diff
@@ -97,6 +99,28 @@ func AWSCleanup(maxConcurrentRequests int, dryRun bool, accessKeyID, accessKey s
 		wg.Wait()
 	}
 
+	// using err to collect both errors as we want to
+	// continue execution if one cleanup fails
+	err = nil
+	errSecureInstances := terminateOrphanedSecureInstances(a, dryRun)
+	// keep going with other cleanup even on error
+	if errSecureInstances != nil {
+		logrus.Errorf("Error in terminating secure instances: %v, continuing other cleanup.", errSecureInstances)
+		err = errSecureInstances
+	}
+
+	errSecurityGroups := searchSGAndCleanup(ctx, a, dryRun)
```

Comment:
I'm not sure if we should share the same context here? It probably doesn't matter that much but I don't get the rationale for doing so here.

Reply:
Like the context isn't passed from up above, but created in this function.

Reply:
I didn't want to change all of the functions from above in this PR, but prepare as much as possible.
As far as I understood that's the idea of the context, that it should always be traversed from top to bottom to be able to hand over cancel, timeouts and other properties…

Reply:
ack